Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -56,52 +66,28 @@ void createSeedsForGroupSycl(
   // Up to the Nth space point, the sum of compatible bottom/top space points.
   // We need these for indexing other vectors later in the algorithm.
   // These are prefix sum arrays, with a leading zero.
-  std::vector<uint32_t> sumBotCompUptoMid(M + 1, 0);
-  std::vector<uint32_t> sumTopCompUptoMid(M + 1, 0);
-  std::vector<uint32_t> sumBotTopCombined(M + 1, 0);
+  // Those will be created with either host or shared memory resource
+  vecmem::vector<uint32_t> sumBotMidPrefix(&resource);
+  sumBotMidPrefix.push_back(0);
+  vecmem::vector<uint32_t> sumTopMidPrefix(&resource);
+  sumTopMidPrefix.push_back(0);
+  vecmem::vector<uint32_t> sumBotTopCombPrefix(&resource);
+  sumBotTopCombPrefix.push_back(0);
 
   // After completing the duplet search, we'll have successfully contructed
   // two bipartite graphs for bottom-middle and top-middle space points.
   // We store the indices of the middle space points of the corresponding
   // edges.
-  std::vector<uint32_t> indMidBotComp;
-  std::vector<uint32_t> indMidTopComp;
-
-  // Number of edges for middle-bottom and middle-top duplet bipartite graphs.
-  uint64_t edgesBottom = 0;
-  uint64_t edgesTop = 0;
-  // Number of possible compatible triplets. This is the sum of the
-  // combination of the number of compatible bottom and compatible top duplets
-  // per middle space point. (nb0*nt0 + nb1*nt1 + ... where nbk is the number
-  // of comp. bot. SPs for the kth middle SP)
-  uint64_t edgesComb = 0;
+  vecmem::vector<uint32_t> indMidBotComp(&resource);
+  vecmem::vector<uint32_t> indMidTopComp(&resource);
```

Comment:
I think these do not need to be `vecmem::vector`-s. They can stay `std::vector`-s, no?

Reply:
I see that the code used to allocate device memory for them in the past... So okay, these may need to be `vecmem::vector` after all. I still didn't look through the full logic... :stuck_out_tongue:

Reply:
Not applicable, I just can't close it. :frowning: